lib/rubocop/cop/style/ternary_parentheses.rb

```diff
@@ -203,16 +201,46 @@ def correct_parenthesized(corrector, condition)
           # If we remove the parentheses, we need to add a space or we'll
           # generate invalid code.
           corrector.insert_after(condition.loc.end, ' ') unless whitespace_after?(condition)
+
+          if (send_node = condition_send_node(condition)) && send_node_args_need_parens?(send_node)
+            parenthesize_condition_arguments(corrector, send_node)
+          end
         end
 
         def correct_unparenthesized(corrector, condition)
           corrector.wrap(condition, '(', ')')
         end
 
+        def parenthesize_condition_arguments(corrector, send_node)
+          range_start = send_node.defined_type? ? send_node.loc.keyword : send_node.loc.selector
+          opening_range = range_start.end.join(send_node.first_argument.source_range.begin)
+
+          corrector.replace(opening_range, '(')
+          corrector.insert_after(send_node.last_argument, ')')
+        end
+
         def whitespace_after?(node)
           last_token = processed_source.last_token_of(node)
           last_token.space_after?
         end
+
+        def condition_send_node(condition)
+          return condition if condition.call_type? || condition.defined_type?
+          return nil unless condition.begin_type? && condition.child_nodes.length == 1
+
+          first_child = condition.child_nodes.first
+          first_child if first_child.call_type? || first_child.defined_type?
+        end
+
+        def send_node_args_need_parens?(send_node)
+          return false unless send_node.call_type? || send_node.defined_type?
+
+          send_node.arguments.any? &&
+            !send_node.parenthesized? &&
+            (send_node.dot? ||
+              send_node.safe_navigation? ||
+              unparenthesized_method_call?(send_node))
```

Comment:
Can you refactor to the following?

```suggestion
          return false unless send_node.call_type? || send_node.defined_type?
          return false if send_node.arguments.none? || send_node.parenthesized?

          send_node.dot? || send_node.safe_navigation? || unparenthesized_method_call?(send_node)
```